baseclasses/utils/tecplotIO.py

```diff
@@ -0,0 +1,1638 @@
+            The file handle.
+        """
+        # Write the zone header
+        zoneString = f'ZONE T="{self.zone.name}"'
+        zoneString += f", DATAPACKING={self.datapacking}"
+
+        # Write the node and element information
+        zoneString += f", NODES={np.multiply.reduce(self.zone.shape):d}"
+        zoneString += f", ELEMENTS={self.zone.nElements:d}"
+        zoneString += f", ZONETYPE={self.zone.zoneType.name}"
+
+        # Write the strand ID and solution time
+        if self.zone.strandID != StrandID.STATIC and self.zone.strandID != StrandID.PENDING:
+            # ASCII format does not support the -1 or -2 strand IDs
+            # So we only write the strand ID if it is not -1
+            zoneString += f", STRANDID={self.zone.strandID}"
+
+        zoneString += f", SOLUTIONTIME={self.zone.solutionTime}\n"
+
+        handle.write(zoneString)
+
+    def writeFooter(self, handle: TextIO):
+        """Write the zone footer to the file. This includes the connectivity information.
+
+        Parameters
+        ----------
+        handle : TextIO
+            The file handle.
+        """
+        connectivity = self.zone.connectivity + 1
+        # Get the max characters in the connectivity
+        maxChars = len(str(connectivity.max()))
+
+        np.savetxt(handle, connectivity, fmt=f"%{maxChars}d")
+
+        handle.write("\n")
+
+
+class TecplotWriterASCII:
+    def __init__(
+        self,
+        title: str,
+        zones: List[TecplotZone],
+        datapacking: Literal["BLOCK", "POINT"],
+        precision: Literal["SINGLE", "DOUBLE"],
+    ) -> None:
+        """Writer for Tecplot files in ASCII format.
+
+        Parameters
+        ----------
+        title : str
+            The title of the Tecplot file.
+        zones : List[TecplotZone]
+            A list of Tecplot zones to write.
+        datapacking : Literal["BLOCK", "POINT"]
+            The data packing format. BLOCK is row-major, POINT is column-major.
+        precision : Literal["SINGLE", "DOUBLE"]
+            The floating point precision to write the data.
+        """
+        self.title = title
+        self.zones = zones
+        self.datapacking = DataPacking[datapacking].name
+        self.precision = DataPrecision[precision].name
+        self._validateVariables()
+
+    def _validateVariables(self) -> None:
+        """Check that all zones have the same variables."""
+        if not all(set(self.zones[0].variables) == set(zone.variables) for zone in self.zones):
+            raise ValueError("All zones must have the same variables.")
+
+    def _writeVariables(self, handle: TextIO) -> None:
+        """Write the variable names to the file.
+
+        Parameters
+        ----------
+        handle : TextIO
+            The file handle.
+        """
+        variables = [f'"{var}"' for var in self.zones[0].variables]
+        variableString = ", ".join(variables)
+        handle.write(f"VARIABLES = {variableString}\n")
```

Comment:
This writes all variable values in a single line, which means if you have too much data tecplot won't open it

![image](https://github.com/user-attachments/assets/74c5d878-c1a7-48a7-bbb1-2b3ebc1395c8)

The data format guide says the max line length is 4000 characters

Reply:
I'll investigate how to define multi-line variables for the ASCII output. I missed the line length requirement. Do you know if this is for data sections as well? Or just the header/metadata information?

Reply:
Sorry, by variable values, I mean the data, that's what's causing the issue, not the header.

Reply:
Ah okay. I'm not sure how to work around this because Tecplot considers a single line a row of data. Is it a bad idea to throw an error and recommend the binary format if the data is too large to fit in ASCII format?

Reply:
I'm almost certain you can put the data across multiple lines

Reply:
e.g
![image](https://github.com/user-attachments/assets/167fd13d-5e00-4af9-ac76-b655462b0512)

Reply:
Maybe I'm just confused, but in that screenshot your text editor is wrapping the lines so it's not actually variables across multiple lines it's just visually wrapping a single line for a single variable.

Reply:
Hmmm, I'm not sure, the number of values on each line doesn't correspond to the number of variables or the number of points in the mesh

Reply:
I confirmed that if the dimensionality of the data is large you cannot write the ASCII file in the `BLOCK` data packing format because it will exceed the line length. You can only write large datasets with the `POINT` data packing option. I will add add logic that will print a warning when this happens and automatically switch to the `POINT` data packing.

This only affects ASCII files. The binary `.plt` files are only written in the block format and there is no line length specifications.

Reply:
I'm 99.9% sure that you can use the BLOCK format for data of any dimensionality, tecplot doesn't seem to give any special meaning to line breaks in ASCII files so you can arbitrarily put line breaks anywhere in the data in order to stay below the line length limit.

![image](https://github.com/user-attachments/assets/401cb7f5-61e4-4539-ae0c-44ce77889f5e)

Reply:
I added custom readers and writers for the block format that adhere to the 4000 max character/line limit. I tested this works with Tecplot and no longer gives the line length error. I also added an assert check for ASCII file line lengths in the tests.

Currently the reader expects the separator to be a space only and does not ignore blank lines. I will create a new reader for the point data packing and then make both the point and block data packing readers work with any of the listed separators and ignore blank lines.

We do not support the repeated number syntax. Should we support this capability for ASCII files?